### DISCUSSION TITLE
Vim file types `asciidoc` and `asciidoctor`

### DISCUSSION THREAD

**Discussion by refaelsh** [w0rp/ale#4511] [General]
Hi all,

All the ASciidoc tools/linters/fixers that are listed here: [ALE Supported Languages and Tools](https://github.com/dense-analysis/ale/blob/master/supported-tools.md), only work when the file type that detected by Vim is `asciidoc`, but I would prefer them to work also when the file type is `asciidoctor`.

Thanks all.

P.S.
Currently, in my Neovim, if I open a file with the `*.adoc` extension, it is recognized by Vim as file type `asciidoctor`.

P.P.S.
How does Vim knows if its a AsciiDoctor file or an AsciiDoc file?

**Comment by hsanson** [comment]
@refaelsh add this to your configuration:

```
let g:ale_linter_aliases = {
      \ 'asciidoctor': 'asciidoc'
      \}
```

This will make all tools that work for asciidoc also work for asciidoctor.

P.S. file type is probably set by some plugin you have. In my case is vim-asciidoctor that has a ftdetect script that set asciidoctor file type to any file with adoc or asciidoc extension.

P.P.S. it doesn't. again some plugin with a ftdetect script must be setting the filetype. Normally based on file extension.

**Comment by refaelsh** [comment]
Thank you very much for the answer, I understand now.
